### ISSUE TITLE
Problema de compatibilidade GuiceProvider e Enviroment

### ISSUE THREAD

**Issue by MAAARKIN** [garcia-jj/vraptor-plugin-hibernate4#17]
Foi constatado ao utilizar o Guice como injetor junto do Enviroment, que o ConfigurationCreator tenta efetuar o seguinte trecho de código:

Linha 62, Environment env = container.instanceFor(Environment.class);

e recebe o seguinte error: http://pastebin.com/ChBdDkZm

Pelo que ví, junto com o (Rogério Freisleben Lacerda) a solução seria essa:
http://pastebin.com/KMPggDcM

**Comment by garcia-jj** [comment]
A única diferença no código é que no atual pegamos o Environment apenas se ele estiver no classpath. E na sugestão ele injeta via construtor. Isso causará incompatibilidade com quem não usa o environment, estourando um `ClassNotFoundException`.

Esse código de sugestão funcionou?

**Comment by MAAARKIN** [comment]
como mencionado, estou usando enviroment e com isso funcionou, criei o pacote e a classe com esse codigo e funcionou, mais bem lembrado, se nao usar enviroment provavelmente resultará em uma exception.

**Comment by MAAARKIN** [comment]
vou tentar chegar a outra solução, no momento estou usando o codigo "sobrescrito" em meu projeto para adiantar, mais vou tentar outras soluções, enfim da forma que está o error persiste.

**Comment by garcia-jj** [comment]
Certo. Obrigado pelo reporte. Vou fazer alguns testes e ver uma solução legal.

Se você quiser fazer a alteração, envie um pull request com o código.

**Comment by MAAARKIN** [comment]
tudo bem, se eu encontrar alguma solução para evitar a injeção do
enviroment via construtor farei, no mais, fica o report.

Em 21 de fevereiro de 2014 10:55, Otávio Garcia
notifications@github.comescreveu:

**Comment by MAAARKIN** [comment]
garcia, atualmente para testar estou criando um package e classe idêntica e modificando a linha necessária, existe alguma outra forma de fazer esse teste? por exemplo, efetuando o fork do projeto e importar esse projeto no meu eclipse, tem como chamar esse projeto pelo maven?

**Comment by dtelaroli** [comment]
Estou tentando remover o Spring de um projeto e estou tendo o mesmo erro.
Algum progresso?

Para usar diretamente importa o projeto no eclipse como um projeto maven.
Mantenha ele aberto.
Adiciona a versão desta dependência no seu pom, que neste caso é 1.5.1-SNAPSHOT
